src/openrct2/GameStateSnapshots.cpp

```diff
@@ -132,8 +133,9 @@ struct GameStateSnapshots final : public IGameStateSnapshots
 
     virtual void Capture(GameStateSnapshot_t& snapshot) override final
     {
-        // TODO refactor to not use this as a proxy for getting a pointer to the sprite array
-        snapshot.SerialiseSprites(get_sprite(0), MAX_SPRITES, true);
+        snapshot.SerialiseSprites(
+            [](const size_t index) -> rct_sprite& { return *reinterpret_cast<rct_sprite*>(GetEntity(index)); }, MAX_SPRITES,
```

Comment:
If this lambda ever tries to dereference a `nullptr` it will crash immediately.

Reply:
Hmm good point.

Reply:
Its never going to happen due to the limits involved but I've now changed it so that it will log an error when it gets to the state that can never happen.